finagle-thrift/src/main/scala/com/twitter/finagle/thrift/Protocols.scala

```diff
@@ -78,6 +78,17 @@ object Protocols {
     binaryFactory(statsReceiver = statsReceiver)
   }
 
+  /**
+   * Multiplex the protocolFactory with the serviceName.
```

Comment:
explain why would you use this? i'm guessing its when you have a client that is talking to a multiplexed backend service?